src/app/modules/main/wallet_section/networks/model.nim

```diff
@@ -76,191 +72,130 @@ QtObject:
+      result = newQVariant(item.blockExplorerURL)
     of ModelRole.NativeCurrencyName:
-      result = newQVariant(item.getNativeCurrencyName())
+      result = newQVariant(item.nativeCurrencyName)
     of ModelRole.NativeCurrencySymbol:
-      result = newQVariant(item.getNativeCurrencySymbol())
+      result = newQVariant(item.nativeCurrencySymbol)
     of ModelRole.IsTest:
-      result = newQVariant(item.getIsTest())
+      result = newQVariant(item.isTest)
     of ModelRole.IsEnabled:
-      result = newQVariant(item.getIsEnabled())
+      result = newQVariant(item.enabled)
     of ModelRole.IconUrl:
-      result = newQVariant(item.getIconURL())
+      result = newQVariant(item.iconURL)
     of ModelRole.ShortName:
-      result = newQVariant(item.getShortName())
+      result = newQVariant(item.shortName)
     of ModelRole.ChainColor:
-      result = newQVariant(item.getChainColor())
+      result = newQVariant(item.chainColor)
     of ModelRole.EnabledState:
-      result = newQVariant(item.getEnabledState().int)
+      result = newQVariant(item.enabledState.int)
 
   proc rowData*(self: Model, index: int, column: string): string {.slot.} =
-    if (index >= self.items.len):
+    if (index >= self.rowCount()):
       return
-    let item = self.items[index]
+    let item = self.delegate.getFlatNetworksList()[index]
     case column:
-      of "chainId": result = $item.getChainId()
-      of "nativeCurrencyDecimals": result = $item.getNativeCurrencyDecimals()
-      of "layer": result = $item.getLayer()
-      of "chainName": result = $item.getChainName()
-      of "rpcURL": result = $item.getRpcURL()
-      of "blockExplorerURL": result = $item.getBlockExplorerURL()
-      of "nativeCurrencyName": result = $item.getNativeCurrencyName()
-      of "nativeCurrencySymbol": result = $item.getNativeCurrencySymbol()
-      of "isTest": result = $item.getIsTest()
-      of "isEnabled": result = $item.getIsEnabled()
-      of "iconUrl": result = $item.getIconURL()
-      of "chainColor": result = $item.getChainColor()
-      of "shortName": result = $item.getShortName()
-      of "enabledState": result = $item.getEnabledState().int
-
-  proc setItems*(self: Model, items: seq[Item]) =
+      of "chainId": result = $item.chainId
+      of "nativeCurrencyDecimals": result = $item.nativeCurrencyDecimals
+      of "layer": result = $item.layer
+      of "chainName": result = $item.chainName
+      of "rpcURL": result = $item.rpcURL
+      of "blockExplorerURL": result = $item.blockExplorerURL
+      of "nativeCurrencyName": result = $item.nativeCurrencyName
+      of "nativeCurrencySymbol": result = $item.nativeCurrencySymbol
+      of "isTest": result = $item.isTest
+      of "isEnabled": result = $item.enabled
+      of "iconUrl": result = $item.iconURL
+      of "chainColor": result = $item.chainColor
+      of "shortName": result = $item.shortName
+      of "enabledState": result = $item.enabledState.int
+
+  proc modelUpdated*(self: Model) =
```

Comment:
If this proc does nothing else than sending begin/end reset model then maybe more appropriate name would be `refreshModel` and also I would emit `countChanged` signal at the end, just for potential bindings.

Reply:
I thought sending begin and endResetModel would eventually cause the count to be updated no? or have you noticed otherwise?

Reply:
No, I haven't tested, just saying, cause count is a separate property and has its signal. If it was reevaluated on the qml side, that's because of some bindings, but if we're using the count property on some other place that is not bound to the model change, then it won't be reevaluated, that's why we need to emit it.

Reply:
so tried like this
    ```
property var flatNetworksCount: networksModule.flatNetworks.count
    onFlatNetworksCountChanged: console.error("onFlatNetworksCountChanged >>>>>>>>>>>>>>>>>")

```
and on modelReset the countChanged signal was also hit. So i guess it not needed separately then.

Reply:
That depends on where you've added this `property var flatNetworksCount: networksModule.flatNetworks.count` property. I am pretty sure it was triggered because of other bindings. If you for example add it to the main app qml it won't be triggered, cause if you look at the code, there is no place that emits `countChange` signal.

Reply:
you are right, Ive added the countchanged to the next PR inline for this. I dont know why I did not understand that first lol